packages/odyssey-react-mui/src/labs/TopNav/TopNav.tsx

```diff
@@ -0,0 +1,180 @@
+/*!
+ * Copyright (c) 2022-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import styled from "@emotion/styled";
+import { memo, type ReactElement } from "react";
+import { useTranslation } from "react-i18next";
+
+import type { HtmlProps } from "../../HtmlProps";
+import { QuestionCircleIcon, SettingsIcon } from "../../icons.generated";
+import { Link } from "../../Link";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import { UserProfile, UserProfileProps } from "./UserProfile";
+import { TopNavLinkProps } from "./TopNavLink";
+import { TopNavLinksList } from "./TopNavLinksList";
+
+export const TOP_NAV_HEIGHT_TOKEN = "Spacing9";
+
+export type TopNavProps = {
+  /**
+   *  Pass in a SearchField component with the variant="filled" prop set
+   */
+  SearchFieldComponent?: ReactElement;
+  /**
+   * Nav links in the top nav
+   */
+  topNavLinkItems: TopNavLinkProps[];
+  /**
+   * Pass in an additional component like `Button` that will be displayed after the nav link items
+   */
+  AdditionalNavItemComponent?: ReactElement;
+  /**
+   * URL to settings page.
+   */
+  settingsPageHref?: string;
+  /**
+   * URL to the help page.
+   */
+  helpPageHref?: string;
+  /**
+   * Displays user account info
+   */
+  userProfile?: UserProfileProps;
```

Comment:
@KevinGhadyani-Okta this should be a component that other teams will have to pass in. Our initial thought was to get the props instead. Since each team has a different menu items inside the User Profile (dropdown), we should just accept a component (like the SearchFieldComponent above) and slot it in.